posthog/models/cohort/sql.py

```diff
@@ -52,7 +47,7 @@
 # Version filtering is not necessary as only positive rows of the latest version will be selected by sum(sign) > 0
 
 GET_PERSON_ID_BY_PRECALCULATED_COHORT_ID = """
-SELECT person_id FROM cohortpeople WHERE team_id = %(team_id)s AND cohort_id = %({prepend}_cohort_id_{index})s GROUP BY person_id, cohort_id, team_id, version HAVING sum(sign) > 0
+SELECT DISTINCT person_id FROM cohortpeople WHERE team_id = %(team_id)s AND cohort_id = %({prepend}_cohort_id_{index})s AND version = %(version)s
```

Comment:
This will resolve [this](https://posthog.slack.com/archives/C045L1VEG87/p1680686797605759)

Context: The query above is trying to group over millions of rows unnecessarily which is causing a memory error